bundles/org.eclipse.e4.ui.progress/src/org/eclipse/e4/ui/progress/internal/ProgressServiceCreationFunction.java

```diff
@@ -16,8 +16,12 @@
 
 import org.eclipse.e4.core.contexts.ContextFunction;
 import org.eclipse.e4.core.contexts.ContextInjectionFactory;
+import org.eclipse.e4.core.contexts.IContextFunction;
 import org.eclipse.e4.core.contexts.IEclipseContext;
+import org.osgi.service.component.annotations.Component;
 
+@Component(service = IContextFunction.class, property = {
+		"service.context.key=org.eclipse.e4.ui.progress.IProgressService" })
```

Comment:
We might want to define a component property type for this @ServiceContectKey(Class)) this will avoid needing to manually set such property

Reply:
Yes that would be handy here. Proposed it in
- https://github.com/eclipse-platform/eclipse.platform/pull/1577